source/extensions/filters/http/common/jwks_fetcher.cc

```diff
@@ -75,6 +114,8 @@ class JwksFetcherImpl : public JwksFetcher,
         if (jwks->getStatus() == google::jwt_verify::Status::Ok) {
           ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: succeeded", __func__, uri_->uri());
           receiver_->onJwksSuccess(std::move(jwks));
+          reset();
+          return;
         } else {
           ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: invalid jwks", __func__, uri_->uri());
           receiver_->onJwksError(JwksFetcher::JwksReceiver::Failure::InvalidJwks);
```

Comment:
This onJwksError() should not be called if we are going to retry.

Reply:
done.